deploy-to-aks/action.yml

```diff
@@ -0,0 +1,147 @@
+name: Deploy to AKS environment
+description: Deploy a docker image to an AKS environment
+
+inputs:
+  azure-credentials:
+    required: true
+  environment:
+    description: Environment to deploy to
+    required: true
+  github-token:
+    description: Default Github token retrieved via secrets.GITHUB_TOKEN or PAT with permission to the repository (Required)
+    required: true
+  pr-number:
+    description: PR number for the review app
+    required: false
+  sha:
+    description: Commit sha to be deployed
+    required: true
+  slack-webhook:
+    required: false
+  db-seed:
+    description: run seed command after a review deployment
+    type: boolean
+    required: false
+    default: false
+  smoke-test:
+    description: Enable smoke test after deployment
+    type: boolean
+    required: false
+    default: false
+  healthcheck:
+    description: Health check path
+    required: false
+  terraform-url-output:
+    description: terraform output that contains the app urls
+    required: false
+    default: 'external_urls'
+  terraform-base:
+    description: path to the terraform files
+    required: false
+    default: 'terraform/application'
+  terraform-provider:
+    description: file containing terraform provider
+    required: false
+    default: 'terraform.tf'
+
+runs:
+  using: composite
+  steps:
+    - name: Set Environment variables
+      id: set_env_var
+      shell: bash
+      run: |
+        terraform_version=$(awk '/{/{f=/^terraform/;next}f' ${{ inputs.terraform-base }}/${{ inputs.terraform-provider }} | grep -o [0-9\.]*)
+        echo "TERRAFORM_VERSION=$terraform_version" >> $GITHUB_ENV
+        echo "GCP_PROJECT_ID=$(make -s printvar-GCP_PROJECT_ID || true)" >> $GITHUB_ENV
+        echo "GCP_WIP=$(make -s printvar-GCP_WIP || true)" >> $GITHUB_ENV
+
+    - name: Use Terraform ${{ env.TERRAFORM_VERSION }}
+      uses: hashicorp/setup-terraform@v3
+      with:
+        terraform_version: ${{ env.TERRAFORM_VERSION }}
+        terraform_wrapper: false
+
+    - uses: azure/login@v2
+      if: ${{ ( inputs.db-seed == 'true' && inputs.environment == 'review' ) }}
+      with:
+        creds: ${{ inputs.azure-credentials }}
+
+    - uses: google-github-actions/auth@v2
+      if: ${{ ( env.PROJECT_ID != '' && env.WIP != '' ) }}
+      with:
+        project_id: ${{ env.PROJECT_ID }}
```

Comment:
project_id may be optional actually: https://github.com/google-github-actions/auth?tab=readme-ov-file#inputs-miscellaneous
we can try without it